### PR TITLE
rbd: disk usage now includes all snapshots by default

### PR DESCRIPTION
**Pull request by dillaman**
Added a new "--from-snap" optional to restrict the calculation to
specific subsets of snapshots and the HEAD revision.

Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

File: src/tools/rbd/action/DiskUsage.cc [hunk 1]
```diff
@@ -152,6 +153,7 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
       continue;
     }
 
+    bool found_from_snap = (from_snapname == nullptr);
```
Comment (jdurgin) [inline src/tools/rbd/action/DiskUsage.cc]: !=?

File: src/tools/rbd/action/DiskUsage.cc [hunk 2]
```diff
@@ -167,7 +169,8 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
         goto out;
       }
 
-      if (imgname == NULL || (snapname != NULL && snap->name == snapname)) {
+      if (imgname == nullptr || found_from_snap ||
+         (found_from_snap && snapname != nullptr && snap->name == snapname)) {
         r = compute_image_disk_usage(*name, snap->name, last_snap_name,
```
Comment (jdurgin) [inline src/tools/rbd/action/DiskUsage.cc]: from_snap instead of last_snap_name? and no need to get last_snap_name anymore?

Comment (jdurgin) [inline src/tools/rbd/action/DiskUsage.cc]: nvm, making sense now

**Comment by jdurgin** [comment]
lgtm, it'd be good to have a simple test in qa/workunits/copy.sh or similar.

File: src/tools/rbd/action/DiskUsage.cc (hunk 1 above)
Comment (dillaman) [inline src/tools/rbd/action/DiskUsage.cc]: Avoiding the need to test "from_snapname == nullptr" two times below by just combing it into the single bool -- if we aren't restricting the starting snapshot, might as well treat it as we've found the starting snapshot.

**Pull request merged**